libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -426,6 +432,65 @@ bool ESP8266WiFiGenericClass::forceSleepWake() {
     return false;
 }
 
+/**
+ * Get listen interval of maximum sleep level for modem sleep and light sleep.
+ * @return interval
+ */
+uint8_t ESP8266WiFiGenericClass::getListenInterval () {
+    return wifi_get_listen_interval();
+}
+
+/**
+ * Set listen interval of maximum sleep level for modem sleep and light sleep.
+ * PREFER using setSleepMode()
+ * datasheet:
+   It only works when sleep level is set as MAX_SLEEP_T
+   It should be called following the order:
+     wifi_set_sleep_level(MAX_SLEEP_T)
+     wifi_set_listen_interval
+     wifi_set_sleep_type
+   forum: https://github.com/espressif/ESP8266_NONOS_SDK/issues/165#issuecomment-416121920
+   default value seems to be 3 (as recommended by https://routerguide.net/dtim-interval-period-best-setting/)
+ */
+void ESP8266WiFiGenericClass::setListenInterval (uint8_t listenInterval) {

```

Comment:
Does it make sense to have a setter just for this when the call must be part of a specific call sequence? It would be trivialfor a user to call it in wrong order.